### PR TITLE
EA-945: implemented server limits for file uploads

### PR DESCRIPTION
**Pull request by p4ddy1**
Implemented ability to limit the size of a file upload utilizing a unique token that needs to be requested from the server before the upload starts. Merge the API Project first!

### CODE REVIEW

File: src/i18n.json [hunk 1]
```diff
@@ -79,7 +79,11 @@
         "app.Upload.UploadButton": "Upload file",
         "app.Upload.Delete": "Delete",
         "app.Upload.OneFileHint": "You may only upload one file!",
-        "app.Upload.FileTooLarge": "This file is too large. Please do not upload files larger than 10 GB",
+        "app.Upload.FileTooLarge": "This file is too large.",
+        "app.Upload.FileSize": "The maximum file size is: ",
+        "app.Upload.Quota": "You have already used ",
+        "app.Upload.QuotaOf": " of ",
+        "app.Upload.QuotaUsed": "",
```
Comment (BrandtM) [inline src/i18n.json]: This mustn't be empty. Please insert at least a space bar, otherwise the translation key is shown.

File: src/index.js [hunk 2]
```diff
@@ -30,6 +30,11 @@ const languageWithoutRegionCode = language.toLowerCase().split(/[_-]+/)[0];
 // Try full locale, try locale without region code, fallback to 'en'
 const messages = localeData[languageWithoutRegionCode] || localeData[language] || localeData.en;
 
+if (configuration.api.substr(0, 5) !== 'https') {
```
Comment (BrandtM) [inline src/index.js]: Doesn't make any sense. Either show an error and don't render the page or redirect to https. But don't just console.error here if https isn't used.

File: src/Component/ResumableDropzone/ReactResumableUploader.js [hunk 3]
```diff
@@ -19,6 +19,8 @@ export default class ReactResumableUploader extends Component {
         generateUniqueIdentifier: PropTypes.func,
         onFileRemoved: PropTypes.func,
         getFileListCallback: PropTypes.func,
+        uploadToken: PropTypes.string,
```
Comment (BrandtM) [inline src/Component/ResumableDropzone/ReactResumableUploader.js]: Is this upload token set anywhere?

File: src/i18n.json (hunk 1 above)
Comment (p4ddy1) [reply src/i18n.json]: Fixed

File: src/index.js (hunk 2 above)
Comment (p4ddy1) [reply src/index.js]: Removed it. I think this happend during the move from GitLab to GitHub.

File: src/Component/ResumableDropzone/ReactResumableUploader.js (hunk 3 above)
Comment (p4ddy1) [reply src/Component/ResumableDropzone/ReactResumableUploader.js]: You are right. I set it using the query parameters. Removed.

**Pull request merged**